ctp2_code/ai/strategy/scheduler/scheduler.cpp

```diff
@@ -631,6 +631,7 @@ void Scheduler::Match_Resources(const bool move_armies)
 
 	sint32 committed_agents = 0;
 	sint32 total_agents     = m_agents.size();
+	bool needToGoBackOne = false;
```

Comment:
Can this be solved without adding an extra variable? It is quiet hard to keep track of all these cases and in most cases it is not needed when the algorithm is changed a little.

Reply:
I think the approach can be changed by iterating through the list in a different fashion - this will hopefully mean any extra variables will be local to the loop.